### PR TITLE
[fix](Nereids) support complex literal cast in fe

### PR DESCRIPTION
**Pull request by morrySnow**
## Proposed changes

Issue Number: close #xxx

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by morrySnow** [comment]
run buildall

**Comment by morrySnow** [comment]
run buildall

**Comment by morrySnow** [comment]
run buildall

**Comment by morrySnow** [comment]
run buildall

**Comment by morrySnow** [comment]
run p0

**Review by starocean999 — APPROVED** [review]

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/literal/StructLiteral.java [hunk 1]
```diff
@@ -42,19 +44,48 @@ public StructLiteral() {
...
+                fields.size(), ((StructType) dataType).getFields().size());
     }
 
     @Override
     public List<Literal> getValue() {
         return fields;
     }
 
+    @Override
+    protected Expression uncheckedCastTo(DataType targetType) throws AnalysisException {
```
Comment (amorynan) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/literal/StructLiteral.java:66]: Is here any chance call legacy literal uncheckedCastTo ?

Comment (morrySnow) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/literal/StructLiteral.java:66]: no

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/literal/StructLiteral.java [hunk 2]
```diff
@@ -89,18 +120,29 @@ public String toString() {
 
     @Override
     public String toSql() {
-        return "{" + fields.stream().map(Literal::toSql).collect(Collectors.joining(",")) + "}";
+        StringBuilder sb = new StringBuilder();
```
Comment (amorynan) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/literal/StructLiteral.java:123]: this func is same with legacyliteral toSqlImpl ? maybe legacyliteral impl is wrong

Comment (morrySnow) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/literal/StructLiteral.java:123]: will rewrite them later

**Review by amorynan — APPROVED** [review]
LGTM

**Pull request merged**